### DISCUSSION TITLE
Would like to grab data from this pdf but not a defined table and text sometimes leaks over to other columns, how do I get all the data?

### DISCUSSION THREAD

**Discussion by MatinQurban** [jsvine/pdfplumber#1164] [Ask for help with specific PDFs]
![image_2024-07-01_180916478](https://github.com/jsvine/pdfplumber/assets/61672425/740899dc-2e4a-4a1a-9a01-2e6fa439a5c8)

`table_settings = {
    "vertical_strategy": "text",
    "horizontal_strategy": "lines",
    "snap_tolerance": 3,
}
im.reset().debug_tablefinder(table_settings)`

<img width="561" alt="image" src="https://github.com/jsvine/pdfplumber/assets/61672425/393bec7a-3b0a-43de-83be-c1ffe90a3a76">

Currently I am going line by line and using each elements relation to everything else on the line to 'guess' where some of the data goes, however sometimes it will not work as the data is very hard to distinguish. For example labor hours and paint hours are very similar values and I cannot simply search for decimal values <10.0 because it would apply to both of them.
Is there a way to get only the labor and paint hours since those two columns seem to be grouped together properly?

**Comment by jsvine** [comment]
This seems like a type of PDF where you might have more luck first identifying the bounding boxes of the page's core sections (using either the bold text in `TRUNK LID`, etc., or the horizontal lines that seem to divide the sections), and then using `page.crop(...)` to iterate over those chunks.

Alternatively, you may have more luck using `page.extract_text(layout=True, x_density=...)`, and then using plain Python and regex to parse each line based on its context.

**Reply by MatinQurban** [reply]
Thank you, I ended up going with the latter
